src/script/arch-update.sh

```diff
@@ -195,22 +200,22 @@ icon_dir="/usr/share/icons/${name}"
 
 # Definition of the icon_checking function: Change icon to "checking"
 icon_checking() {
-	cp -f "${icon_dir}/${name}_checking.svg" "${icon_dir}/${name}.svg" || exit 3
+	echo "${name}_checking" > "${ARCH_UPDATE_STATE}"
```

Comment:
If we are able to use the already existing $statedir mechanism, we can do something like `echo "${name}_checking" > "${statedir}/state"` (or `${statedir}/**current**_state` to comply with the already used "last/current" prefix for state related files).

Reply:
went with `${statedir}/state`